aws/product/awsec2product.go

```diff
@@ -0,0 +1,191 @@
+package product
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit2/db"
+	"github.com/trackit/trackit2/models"
+)
+
+var (
+	// storeAttribute contains the reference between
+	// a field name and the way to store its value.
+	storeAttribute = map[string]func(string, interface{}, *models.AwsProductEc2, jsonlog.Logger){
+		"instanceType":       storeAttributeGenericString,
+		"currentGeneration":  storeAttributeGenericBool,
+		"vcpu":               storeAttributeGenericInt,
+		"memory":             storeAttributeGenericString,
+		"storage":            storeAttributeGenericString,
+		"networkPerformance": storeAttributeGenericString,
+		"tenancy":            storeAttributeGenericString,
+		"operatingSystem":    storeAttributeGenericString,
+		"ecu":                storeAttributeGenericInt,
+		"location":           storeAttributeRegion,
+	}
+)
+
+const (
+	// urlEC2Pricing is the URL used by downloadJSON to
+	// fetch the EC2 pricing.
+	urlEC2Pricing = "https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonEC2/current/index.json"
+)
+
+// storeAttributeRegion creates the region in the database
+// if it doesn't exist. Then, the function stores the region's
+// ID in the struct which will be imported in the database.
+func storeAttributeRegion(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	dbAwsRegion, err := models.AwsRegionByPretty(db.Db, value.(string))
+	if err != nil {
```

Comment:
You check if `err != nil` but does it only mean that the region is not created inside the database ?
You do not check if it could be another type of error